### ISSUE TITLE
Cant issue certificate but wont tell me why

### ISSUE THREAD

**Issue by MarvinMuuss** [caddy-dns/googleclouddns#6]
Hi there,
I'm currently trying to implement the googleclouddns as provider for our Caddy setup. For now I do this manually before I try to automate everything.
Unfortunately it fails to solve the dns challenge, but I cant grasp why and help would be highly appreciated.

The setup:
- Caddy server: v2.6.2 h1:wKoFIxpmOJLGl3QXoo6PNbYvGW4xLEgo32GPBEjWL8o= (with googleclouddns included)
- Caddyfile
```
{
        acme_ca https://acme-staging-v02.api.letsencrypt.org/directory
        email me@company.com
        debug
}

my-test-subdomain.company.com {
        tls {
                dns googleclouddns {
                        gcp_project gcp-dns-project
                }
        }
        reverse_proxy 127.0.0.1:8200
        log {
                output file /var/caddy.log
                format console 
        }
}
```

I run `caddy run --config /etc/caddy/Caddyfile` and will attach the output.
The _acme-challenge DNS record is created successfully, I can see the TXT record and a DNS checker also verified it. 
So I cant really tell why it fails. The error message
```
2022/10/19 14:57:52.491 ERROR   http.acme_client        cleaning up solver      {"identifier": "my-test-subdomain.company.com", "challenge_type": "dns-01", "error": "no memory of presenting a DNS record for \"_acme-challenge.my-test-subdomain.company.com\" (usually OK if presenting also failed)"}
2022/10/19 14:57:52.635 ERROR   tls.obtain      could not get certificate from issuer   {"identifier": "my-test-subdomain.company.com", "issuer": "acme-staging-v02.api.letsencrypt.org-directory", "error": "[my-test-subdomain.company.com] solving challenges: waiting for solver certmagic.solverWrapper to be ready: timed out waiting for record to fully propagate; verify DNS provider configuration is correct - last error: <nil> (order=https://acme-staging-v02.api.letsencrypt.org/acme/order/72908704/4656280764) (ca=https://acme-staging-v02.api.letsencrypt.org/directory)"}
```
Isn't really telling me much.

I'd really appreciate any support or ideas how to solve this.

Thanks and BR Marvin
[caddy-error.log](https://github.com/caddy-dns/googleclouddns/files/9821882/caddy-error.log)

PS: I've replaced sensitive data like domain and my emailaddress

**Comment by brockwood** [comment]
Interesting. This module only provides updating of DNS records in Cloud DNS and doesn't really do anything else beyond that. With that in mind, I'm working on a fix for another issue so it may be worth trying this again once that is merged. I hope to have a new version of the module released this weekend.

**Comment by MarvinMuuss** [comment]
Hey @brockwood,

thanks for your reply. During the weekend I took another look at it with a rested mind and no hurries of usual working day.

I tried to configure Caddy more explicit and this worked:
```
my.domain {
    tls {
            issuer zerossl {
                email myemail@company.com
                disable_http_challenge
                disable_tlsalpn_challenge
                dns googleclouddns {
                        gcp_project my-dns-project
                }
                resolvers 8.8.8.8
                dns_challenge_override_domain _acme-challenge.my.domain
            }
    }
    reverse_proxy 127.0.0.1:8200
}
```

So when this module is only there for the update of the DNS record I guess this ticket can be closed, since this works fine.

But I have another issue: The TXT record is not deleted after successful challenge. Shall I issue another ticket?

**Comment by brockwood** [comment]
Hi @MarvinMuuss!

I wondered if adding the `resolvers` to the config would help out. I'm glad that works. 

I stumbled upon the TXT record not being cleaned up while taking a look at your issue. I have a PR against this repo's upstream libdns library, libdns/googleclouddns#4, that fixes the TXT cleanup bug. I'm currently testing it out and hope to have it merged soon. Thanks again for reaching out on this; it is appreciated. :smile: I'll leave a note here and close the issue when the upstream PR is merged.

**Comment by almereyda** [comment]
Upstream has been merged. Is this now included here, and can thus be closed?

**Comment by brockwood** [comment]
Yes @almereyda, thanks for pointing that out. It slipped my mind.
